ITMO_FS/hybrid/IWSSr_SFLA.py

```diff
@@ -0,0 +1,342 @@
+                The training input samples.
+            y : array-like, shape (n_samples)
+                The target values.
+
+            Returns
+            -------
+            array-like, shape (sfla_m * sfla_n, ) : array of frogs
+        """
+        frogs_transformed = []
+
+        for frog in frogs:
+            transformed_frog = frog[self.iwssr(X[:, frog], y)]
+            frogs_transformed.append(np.array(transformed_frog))
+
+        return frogs_transformed
+
+    def iwssr(self, X, y):
+        """
+            Performs the IWSSr algorithm for given data.
+
+            Parameters
+            ----------
+            X : array-like, shape (n_samples, n_features)
+                The training input samples.
+            y : array-like, shape (n_samples)
+                The target values.
+
+            Returns
+            -------
+            array-like, shape (n_selected_features) : selected features
+        """
+        weights = su_measure(X, y)
+        order = np.argsort(weights)[::-1]
+        selected_features = np.array([order[0]])
+        best_measure = cross_val_score(self._estimator, X[:, selected_features], y, cv=self.cv,
+                                                    scoring=self.measure_iwssr).mean()
+        best_iteration_set = selected_features
+        for feature in order[1:]:
+            for i in range(len(selected_features)):
+                iteration_features = np.append(np.delete(selected_features, i), feature)
+                iteration_measure = cross_val_score(self._estimator, X[:, iteration_features], y, cv=self.cv,
+                                                    scoring=self.measure_iwssr).mean()
+                if iteration_measure > best_measure:
+                    best_measure = iteration_measure
+                    best_iteration_set = iteration_features
+            added_feature = np.append(selected_features, feature)
+            added_measure = cross_val_score(self._estimator, X[:, added_feature], y, cv=self.cv,
+                                                    scoring=self.measure_iwssr).mean()
+            if added_measure > best_measure:
+                best_measure = added_measure
+                best_iteration_set = added_feature
+            selected_features = best_iteration_set
+        return selected_features
+
+    def create_submemplex(self, fitness_values):
+        """
+            Creates a submemplex from the memplex of frogs.
+
+            Parameters
+            ----------
+            fitness_values : array-like, shape (sfla_n)
+                Fitness values for all frogs in the memplex.
+
+            Returns
+            -------
+            array-like, shape (sfla_q) : indices of frogs in the submemplex
+        """
+        order = np.argsort(fitness_values)[::-1]
+        weights = np.zeros(self.sfla_n)
+        for i in range(self.sfla_n):
+            weights[order[i]] = 2 * (self.sfla_n - i) / (self.sfla_n * (self.sfla_n + 1))
```

Comment:
This could be simplified to (2-2i)/self.sfla_n

Reply:
(2 - 2i) is a negative number for i > 1 and the existing formula always yields positive values, there's a mistake in your math I guess